crates/aptos-rosetta/src/account.rs

```diff
@@ -137,156 +142,198 @@ async fn get_balances(
-                    }*/
-                },
-                _ => {},
-            }
-        }
-
-        // Retrieves the sequence number accordingly
-        // TODO: Sequence number should be 0 if it isn't retrieved probably
-        let sequence_number = if let Some(sequence_number) = maybe_sequence_number {
-            sequence_number
-        } else {
+            ..
+        })) => {
+            // If the account or resource doesn't exist, set the sequence number to 0
+            0
+        },
+        _ => {
+            // Any other error we can't retrieve the sequence number
             return Err(ApiError::InternalError(Some(
                 "Failed to retrieve account sequence number".to_string(),
             )));
-        };
-
-        // Filter based on requested currencies
-        if let Some(currencies) = maybe_filter_currencies {
-            let mut currencies: HashSet<Currency> = currencies.into_iter().collect();
-            // Remove extra currencies not requested
-            balances = balances
-                .into_iter()
-                .filter(|balance| currencies.contains(&balance.currency))
-                .collect();
+        },
+    };
 
-            for balance in balances.iter() {
-                currencies.remove(&balance.currency);
+    // Retrieve staking information (if it applies)
+    // Only non-pool addresses, and non base accounts
+    let mut maybe_operators = None;
+    if !account.is_base_account() && pool_address.is_none() {
+        if let Ok(response) = rest_client
+            .get_account_resource_at_version_bcs(
+                owner_address,
+                "0x1::staking_contract::Store",
+                version,
+            )
+            .await
+        {
+            let store: Store = response.into_inner();
+            maybe_operators = Some(vec![]);
+            for (operator, contract) in store.staking_contracts {
+                // Keep track of operators
+                maybe_operators.as_mut().unwrap().push(operator);
+                match get_stake_balances(
+                    rest_client.as_ref(),
+                    &account,
+                    contract.pool_address,
+                    version,
+                )
+                .await
+                {
+                    Ok(Some(balance_result)) => {
+                        if let Some(balance) = balance_result.balance {
+                            total_requested_balance = Some(
+                                total_requested_balance.unwrap_or_default()
+                                    + u64::from_str(&balance.value).unwrap_or_default(),
+                            );
+                        }
+                        lockup_expiration = balance_result.lockup_expiration;
+                    },
+                    result => {
+                        warn!(
+                                        "Failed to retrieve requested balance for account: {}, address: {}: {:?}",
```

Comment:
This pre-existed, but the idea is to ensure things keep running rather than crash.  Error handling with the systems applied around don't always handle it gently.

In this case, it's around poor formatting, which should not change.  BCS is a bit of a challenge in itself.